### PR TITLE
Add Devconsole.ts, typescript based utility console

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

`Devconsole.ts` is a command-line utility to quickly work with `wallet-core` library.
It is based on `node` `typescript` console, and uses the WASM version of `wallet-core`.

Related to #2422 .

## How to test

```
cd samples/typescript/devconsole.ts
npm install
npm run build
npm run start
```

type:

```
> help()
This is an interactive typescript shell, to work with wallet-core (wasm)
You can use:
...
> w1 = HDWallet.create(256, '')
W {}
> w1.mnemonic()
'round profit immense ... sniff'
> w1.getAddressForCoin(CoinType.cosmos)
'cosmos1kgd25tvkz3jtuee4cl796trc2s0hjsfgqmhmku'
```

Also, building it is added to the sample app CI.

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [x] Create pull request as draft initially, unless its complete.
- [ ] Add tests to cover changes as needed.
- [ ] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description.

If you're adding a new blockchain

- [ ] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain.

### CODE REVIEW

File: .github/workflows/linux-sampleapp-ci.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Linux SampleApp CI
+`name: Linux SampleApps CI
```
Comment (miloserdow) [inline .github/workflows/linux-sampleapp-ci.yml:1]: What's this ` is doing here?

Comment (catenocrypt) [reply .github/workflows/linux-sampleapp-ci.yml]: Accidentally pressed wrong key >-( fixed

File: samples/typescript/devconsole.ts/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        console.log("    HexCoding.encode(pk.data())");
+        console.log("    pubkey = pk.getPublicKeyEd25519()");
+        console.log("    HexCoding.encode(pubkey.data())");
+        console.log("    a1 = AnyAddress.createWithPublicKey(pubkey, coin)");
+        console.log("    a1.description()");
+        console.log();
+    }
+
+    function walletCreate(strength: number = 256, name: string = ''): any {
+        /*
```
Comment (miloserdow) [inline samples/typescript/devconsole.ts/src/index.ts:104]: Let's remove commented-out blocks :)

**Comment by catenocrypt** [comment]
Here's the current output from the CI :)

image

File: samples/typescript/devconsole.ts/package-lock.json [hunk 3]
```diff
@@ -0,0 +1,1491 @@
+{
```
[the remark this answers is not in the document]
Comment (catenocrypt) [reply samples/typescript/devconsole.ts/package-lock.json:1]: Package-lock.json is usually included under version control, see https://duckduckgo.com/?t=ffab&q=should+i+include+package-lock.json+in+git&ia=web

File: samples/typescript/devconsole.ts/.gitignore [hunk 4]
```diff
@@ -0,0 +1,3 @@
+node_modules
+lib
+
```
[the remark this answers is not in the document]
Comment (catenocrypt) [reply samples/typescript/devconsole.ts/.gitignore:3]: Package-lock.json is usually included under version control, see https://duckduckgo.com/?t=ffab&q=should+i+include+package-lock.json+in+git&ia=web

**Review by miloserdow — DISMISSED** [review]

File: .github/workflows/linux-sampleapp-ci.yml [hunk 5]
```diff
@@ -55,6 +55,20 @@ jobs:
...
+        npm run build
+        # note: -e for echo is needed for \n's to work on bash
+        echo -e \
+          "privkey = PrivateKey.createWithData(HexCoding.decode('afeefca74d9a325cf1d6b6911d61a65c32afa8e02bd5e78e2e4ac2910bab45f5')) \n " \
+          "HexCoding.encode(privkey.data()) \n " \
+          "pubkey = privkey.getPublicKeySecp256k1(true) \n " \
+          "HexCoding.encode(pubkey.data()) \n " \
+          "pubkey.verify(HexCoding.decode('673b54a91d87cfb9389e54cc55b1a9343a6eb9f2ea1f449cb19a248a86bb904c1efb109f8cf655c4f510211053c1696e52c75843c5c803fa1b78fe0c263f468201'), HexCoding.decode('0001020304050607080910111213141519171819202122232425262728293031')) \n " \
+          ".exit " \
+          | npm run start
```
Comment (Milerius) [inline .github/workflows/linux-sampleapp-ci.yml:71]: can you wrap it in a script ?

**Review by Milerius — DISMISSED** [review]
LGTM

File: .github/workflows/linux-sampleapp-ci.yml (hunk 5 above)
Comment (catenocrypt) [reply .github/workflows/linux-sampleapp-ci.yml:71]: Best would be to make this into a test, called by `npm run test`

Comment (catenocrypt) [reply .github/workflows/linux-sampleapp-ci.yml]: Input moved to a separate file, invocation moved to `npm run test`.

**Comment by hewigovens** [comment]
Can you add one example how to use protobuf models? the rest looks good, we can add more useful commands later

**Comment by catenocrypt** [comment]
> Can you add one example how to use protobuf models? the rest looks good, we can add more useful commands later

Protobuf example added:  https://github.com/trustwallet/wallet-core/blob/ar/devconsole.ts/samples/typescript/devconsole.ts/samplescripts/protoeth.sampleinput.txt

**Review by hewigovens — APPROVED** [review]

File: samples/typescript/devconsole.ts/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+    console.log(
+        chalk.blue(
+            figlet.textSync('wallet-core', { horizontalLayout: 'full' })
+        )
+    );
+    console.log(chalk.blue('  devconsole.ts, wasm lib'));
+    console.log();
+
+    process.stdout.write("Initializing WalletCore library ...");
+    const WC = await initWasm();
```
Comment (hewigovens) [inline samples/typescript/devconsole.ts/src/index.ts:176]: I opened in VSCode, WC type is `any`, Is there any way to enable the types?

**Review by miloserdow — APPROVED** [review]

**Comment by 09168806659** [comment]
First of all, hello to everyone, I regret that thousands of hyenas are looking for the prey that the lion worked hard for for himself and his family for years, and in the end, due to an accident, he has been lying in a hospital bed for 17 months. 
He sold his two sons to treat him, and wishes and efforts for many years for his two sons for their future and the future of his country, that they become great developers like him, of course, not like many carrion eaters,, yesterday also because of 12 days 
Due to the hospital's lack of money, they said that they want to take away the ventilators and other facilities from her. I am a woman and I am no longer responsible for all these expenses. You can also draw a plan for the carcass of this lion and its 2 sons, but remember that this lion is in trouble. 
He came, so how about the hyena,,,✋️

**Comment by 09168806659** [comment]
I remember him saying that the director of Binance, I see a kind of manliness in him, that he accepted the request of America not to accept the embargoed countries, but the American no longer accepts (and said he is the manager of TrustVault.
the same)
He doesn't care about anyone, he does what he knows is right, but now I see something else??
The manager,, me, his wife ((Rezadars)), that colleague, I know he will be fine,, 🙏 Whatever it is, please open his account and don't let him lose it.
Thank you

**Pull request merged**